dpdk/vr_dpdk_interface.c

```diff
@@ -356,12 +395,58 @@ dpdk_monitoring_if_add(struct vr_interface *vif)
     if (ret != 0)
         return ret;
 
-    /* setup monitoring */
-    dpdk_monitoring_setup(monitored_vif, vif);
+    /* start monitoring */
+    dpdk_monitoring_start(monitored_vif, vif);
 
     return 0;
 }
 
+/* Delete monitoring interface */
+static int
+dpdk_monitoring_if_del(struct vr_interface *vif)
+{
+    unsigned short monitored_vif_id = vif->vif_os_idx;
+    struct vr_interface *monitored_vif;
+
+    RTE_LOG(INFO, VROUTER, "Deleting monitoring vif %u KNI device"
+                " to monitor vif %u\n",
+                vif->vif_idx, monitored_vif_id);
+
+    /* check if vif exist */
+    monitored_vif = vrouter_get_interface(vif->vif_rid, monitored_vif_id);
+    if (!monitored_vif) {
+        RTE_LOG(ERR, VROUTER, "\terror getting vif to monitor: vif %u does not exist\n",
+                monitored_vif_id);
+        return -EINVAL;
+    }
+
+    /* stop monitoring */
+    dpdk_monitoring_stop(monitored_vif, vif);
+    vrouter_put_interface(monitored_vif);
```

Comment:
Do we need to get and put in this function (i.e. can this just be __vrouter_get_interface)?